File: src/main/java/dev/isxander/evergreenhud/elements/impl/ElementText.java
/*
 * Copyright (C) isXander [2019 - 2021]
 * This program comes with ABSOLUTELY NO WARRANTY
 * This is free software, and you are welcome to redistribute it
 * under the certain conditions that can be found here
 * https://www.gnu.org/licenses/gpl-3.0.en.html
 *
 * If you have any questions or concerns, please create
 * an issue on the github page that can be found here
 * https://github.com/isXander/EvergreenHUD
 *
 * If you have a private concern, please contact
 * isXander @ business.isxander@gmail.com
 */

package dev.isxander.evergreenhud.elements.impl;

import dev.isxander.evergreenhud.elements.type.SimpleTextElement;
import dev.isxander.evergreenhud.settings.impl.StringSetting;
import dev.isxander.evergreenhud.elements.ElementData;

public class ElementText extends SimpleTextElement {

    public StringSetting text;

    @Override
    public void initialise() {
        addSettings(text = new StringSetting("Text Element", "Display", "What the value will display.", "Text Element"));
    }

    @Override
    public ElementData metadata() {
        return new ElementData("Custom Text", "Displays a custom message of your liking!", "Advanced");
    }

    @Override
    protected String getValue() {
        return convertColorCodes(text.get());
    }

    private String convertColorCodes(String text) {
        return text.replaceAll("&([a-fA-F0-9])", "\u00A7$1");
    }

    @Override
    public String getDefaultDisplayTitle() {
        return "";
    }

}

File: src/main/kotlin/dev/isxander/evergreenhud/elements/impl/ElementText.kt
/*
 * EvergreenHUD - A mod to improve your heads-up-display.
 * Copyright (c) isXander [2019 - 2021].
 *
 * This work is licensed under the CC BY-NC-SA 4.0 License.
 * To view a copy of this license, visit http://creativecommons.org/licenses/by-nc-sa/4.0
 */

package dev.isxander.evergreenhud.elements.impl

import dev.isxander.evergreenhud.elements.type.SimpleTextElement
import dev.isxander.evergreenhud.utils.elementmeta.ElementMeta
import dev.isxander.settxi.impl.string

@ElementMeta(id = "TEXT", name = "Text Display", category = "Other", description = "Displays custom text of your choosing.")
class ElementText : SimpleTextElement() {
    var text by string("Sample Text") {
        name = "Text"
        category = "Text"
        description = "The text to display."
    }

    init {
        title = ""
    }

    override fun calculateValue(): String = text
}
